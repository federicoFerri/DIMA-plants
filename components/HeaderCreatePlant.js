import React from 'react';
import {Image, Text, SafeAreaView, TouchableOpacity, View } from 'react-native';
import BackButton from '../buttons/BackButton'
import ForwardButton from '../buttons/ForwardButton';

{/*props required
  @backPress: if back arrow is pressed, it launches an event like TouchableOpacity
  @forwardPress: if forward arrow is pressed, it launches an event like TouchableOpacity
*/}
class HeaderCreatePlant extends React.Component {
    forwardPressed = () => () => {
        this.props.forwardPress()
    }
    backPressed = () => () => {
        this.props.backPress()
    }
    render() {
        return (
            <SafeAreaView style={{flexDirection: 'row', justifyContent:'space-between', alignItems:'center', marginTop: 20, marginLeft: 15, marginRight: 15}}>
                <BackButton
                    onPress={this.backPressed()}
                />
                {this.props.isForwardVisible && <ForwardButton
                    onPress={this.forwardPressed()}
                />}
            </SafeAreaView>
        )
    }
}

HeaderCreatePlant.defaultProps = {
    isForwardVisible: true,
  };

export default HeaderCreatePlant;