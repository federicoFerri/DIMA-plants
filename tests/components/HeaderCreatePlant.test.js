import React from 'react';
import renderer from 'react-test-renderer';

import HeaderCreatePlant from '../../components/HeaderCreatePlant'

describe('<HeaderCreatePlant />', () => {
    test('renders correctly', () => {
        const tree = renderer.create(<HeaderCreatePlant/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});