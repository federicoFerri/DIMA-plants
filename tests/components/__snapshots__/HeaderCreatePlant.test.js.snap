// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderCreatePlant /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginLeft": 15,
      "marginRight": 15,
      "marginTop": 20,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 21,
            "resizeMode": "contain",
            "width": 20,
          }
        }
      />
    </View>
  </RCTSafeAreaView>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 27,
            "resizeMode": "contain",
            "width": 32,
          }
        }
      />
    </View>
  </RCTSafeAreaView>
</RCTSafeAreaView>
`;
